man/ocaml.1
.\"**************************************************************************
.\"*                                                                        *
.\"*                                 OCaml                                  *
.\"*                                                                        *
.\"*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           *
.\"*                                                                        *
.\"*   Copyright 1996 Institut National de Recherche en Informatique et     *
.\"*     en Automatique.                                                    *
.\"*                                                                        *
.\"*   All rights reserved.  This file is distributed under the terms of    *
.\"*   the GNU Lesser General Public License version 2.1, with the          *
.\"*   special exception on linking described in the file LICENSE.          *
.\"*                                                                        *
.\"**************************************************************************
.\"
.TH OCAML 1

.SH NAME
ocaml \- The OCaml interactive toplevel

.SH SYNOPSIS
.B ocaml
[
.I options
]
[
.I object-files
]
[
.I script-file
]
.SH DESCRIPTION

The
.BR ocaml (1)
command is the toplevel system for OCaml,
that permits interactive use of the OCaml system through a
read-eval-print loop. In this mode, the system repeatedly reads OCaml
phrases from standard input, then typechecks, compiles and evaluates
them, then prints the inferred type and result value, if any.
End-of-file on standard input terminates
.BR ocaml (1).

Input to the toplevel can span several lines. It begins after the #
(sharp) prompt printed by the system and is terminated by ;; (a
double-semicolon) followed by optional white space and an end of line.
The toplevel input consists in one or several toplevel phrases.

If one or more
.I object-files
(ending in .cmo or .cma) are given on the command line, they are
loaded silently before starting the toplevel.

If a
.I script-file
is given, phrases are read silently from the file, errors printed on
standard error.
.BR ocaml (1)
exits after the execution of the last phrase.

.SH OPTIONS

The following command-line options are recognized by
.BR ocaml (1).
.TP
.B \-absname
Show absolute filenames in error messages.
.TP
.B \-no-absname
Do not try to show absolute filenames in error messages.
.TP
.B \-i-variance
Cause the compiler to print the computed variance of type parameters for
every type declaration.  Without this option, variance is printed only for
private types or types without a manifest (e.g., abstract types).
.TP
.BI \-I " directory"
Add the given directory to the list of directories searched for
source and compiled files. By default, the current directory is
searched first, then the standard library directory. Directories added
with
.B \-I
are searched after the current directory, in the order in which they
were given on the command line, but before the standard library
directory.
.IP
If the given directory starts with
.BR + ,
it is taken relative to the
standard library directory. For instance,
.B \-I\ +compiler-libs
adds the subdirectory
.B compiler-libs
of the standard library to the search path.
.IP
Directories can also be added to the search path once the toplevel
is running with the
.B #directory
directive.
.TP
.BI \-init " file"
Load the given file instead of the default initialization file.
See the "Initialization file" section below.
.TP
.B \-labels
Labels are not ignored in types, labels may be used in applications,
and labelled parameters can be given in any order.  This is the default.
.BI \-log\-format " format"
Use the format
.IR format
for diagnostics (warnings, error messages, REPL evaluation).
.TP
.BI \-log\-version " major.minor"
Emit diagnostics with a format compatible with the version
.IR major.minor "."
.TP
.BI \-log\-file " filename"
Emit diagnostics to the file
.IR filename "."
.TP
.B \-no\-app\-funct
Deactivates the applicative behaviour of functors. With this option,
each functor application generates new types in its result and
applying the same functor twice to the same argument yields two
incompatible structures.
.TP
.B \-noassert
Do not compile assertion checks.  Note that the special form
.B assert\ false
is always compiled because it is typed specially.
.TP
.B \-noinit
Do not load any initialization file.
See the "Initialization file" section below.
.TP
.B \-nolabels
Ignore non-optional labels in types. Labels cannot be used in
applications, and parameter order becomes strict.
.TP
.B \-noprompt
Do not display any prompt when waiting for input.
.TP
.B \-nopromptcont
Do not display the secondary prompt when waiting for continuation lines in
multi-line inputs.  This should be used e.g. when running
.BR ocaml (1)
in an
.BR emacs (1)
window.
.TP
.B \-nostdlib
Do not include the standard library directory in the list of
directories searched for source and compiled files.
.TP
.BI \-open " module"
Opens the given module before starting the toplevel. If several
.B \-open
options are given, they are processed in order, just as if
the statements open! module1;; ... open! moduleN;; were input.
.TP
.BI \-ppx " command"
After parsing, pipe the abstract syntax tree through the preprocessor
.IR command .
The module
.BR Ast_mapper (3)
implements the external interface of a preprocessor.
.TP
.B \-principal
Check information path during type-checking, to make sure that all
types are derived in a principal way.  When using labelled arguments
and/or polymorphic methods, this flag is required to ensure future
versions of the compiler will be able to infer types correctly, even
if internal algorithms change.
All programs accepted in
.B \-principal
mode are also accepted in the
default mode with equivalent types, but different binary signatures,
and this may slow down type checking; yet it is a good idea to
use it once before publishing source code.
.TP
.B \-no\-principal
Do not check principality of type inference.
This is the default.
.TP
.B \-rectypes
Allow arbitrary recursive types during type-checking.  By default,
only recursive types where the recursion goes through an object type
are supported.
.TP
.B \-no\-rectypes
Do no allow arbitrary recursive types during type-checking.
This is the default.
.TP
.B \-safe\-string
Enforce the separation between types
.BR string " and " bytes ,
thereby making strings read-only. This is the default.
.TP
.B \-safer\-matching
Do not use type information to optimize pattern-matching.
This allows to detect match failures even if a pattern-matching was
wrongly assumed to be exhaustive. This only impacts GADT and
polymorphic variant compilation.
.TP
.B \-short\-paths
When a type is visible under several module-paths, use the shortest
one when printing the type's name in inferred interfaces and error and
warning messages.
.TP
.B \-stdin
Read the standard input as a script file rather than starting an
interactive session.
.TP
.B \-strict\-sequence
Force the left-hand part of each sequence to have type unit.
.TP
.B \-no\-strict\-sequence
Left-hand part of a sequence need not have type unit.
This is the default.
.TP
.B \-unboxed\-types
When a type is unboxable (i.e. a record with a single argument or a
concrete datatype with a single constructor of one argument) it will
be unboxed unless annotated with
.BR [@@ocaml.boxed] .
.TP
.B \-no-unboxed\-types
When a type is unboxable  it will be boxed unless annotated with
.BR [@@ocaml.unboxed] .
This is the default.
.TP
.B \-unsafe
Turn bound checking off on array and string accesses (the
.BR v.(i) " and " s.[i]
constructs). Programs compiled with
.B \-unsafe
are therefore slightly faster, but unsafe: anything can happen if the program
accesses an array or string outside of its bounds.
.TP
.B \-unsafe\-string
Identify the types
.BR string " and " bytes ,
thereby making strings writable.
This is intended for compatibility with old source code and should not
be used with new software.
.TP
.B \-version
Print version string and exit.
.TP
.B \-vnum
Print short version number and exit.
.TP
.B \-no\-version
Do not print the version banner at startup.
.TP
.BI \-w " warning\-list"
Enable or disable warnings according to the argument
.IR warning-list .
See
.BR ocamlc (1)
for the syntax of the
.I warning\-list
argument.
.TP
.BI \-warn\-error " warning\-list"
Mark as fatal the warnings described by the argument
.IR warning\-list .
Note that a warning is not triggered (and does not trigger an error) if
it is disabled by the
.B \-w
option.  See
.BR ocamlc (1)
for the syntax of the
.I warning\-list
argument.
.TP
.BI \-color " mode"
Enable or disable colors in compiler messages (especially warnings and errors).
The following modes are supported:

.B auto
use heuristics to enable colors only if the output supports them (an
ANSI-compatible tty terminal);

.B always
enable colors unconditionally;

.B never
disable color output.

The environment variable "OCAML_COLOR" is considered if \-color is not
provided. Its values are auto/always/never as above.

If \-color is not provided, "OCAML_COLOR" is not set and the environment
variable "NO_COLOR" is set, then color output is disabled. Otherwise,
the default setting is
.B auto,
and the current heuristic
checks that the "TERM" environment variable exists and is
not empty or "dumb", and that isatty(stderr) holds.

.TP
.BI \-error\-style " mode"
Control the way error messages and warnings are printed.
The following modes are supported:

.B short
only print the error and its location;

.B contextual
like "short", but also display the source code snippet corresponding
to the location of the error.

The default setting is
.B contextual.

The environment variable "OCAML_ERROR_STYLE" is considered if
\-error\-style is not provided. Its values are short/contextual as
above.

.TP
.B \-warn\-help
Show the description of all available warning numbers.
.TP
.BI \- " file"
Use
.I file
as a script file name, even when it starts with a hyphen (-).
.TP
.BR \-help " or " \-\-help
Display a short usage summary and exit.

.SH INITIALIZATION FILE

When
.BR ocaml (1)
is invoked as a toplevel, it will read phrases from an initialization file
before giving control to the user. This does not happen when running a
script, or when using -e.

The file read is the first found of:
.IP 1.
\fB.ocamlinit\fP in the current directory;
.IP 2.
\fBXDG_CONFIG_HOME/ocaml/init.ml\fP, if \fBXDG_CONFIG_HOME\fP is an absolute
path;
.IP 3.
otherwise, on Unix, \fBHOME/ocaml/init.ml\fP or, on Windows,
\fBocaml\\init.ml\fP under \fBLocalAppData\fP
(e.g. \fBC:\\Users\\Bactrian\\AppData\\Local\\ocaml\\init.ml\fP);
.IP 4.
\fBocaml/init.ml\fP under any of the absolute paths in \fBXDG_CONFIG_DIRS\fP.
Paths in \fBXDG_CONFIG_DIRS\fP are colon-delimited on Unix, and
semicolon-delimited on Windows;
.IP 5.
if \fBXDG_CONFIG_DIRS\fP contained no absolute paths,
\fB/usr/xdg/ocaml/init.ml\fP on Unix or, \fBocaml\\init.ml\fP under any of
\fBLocalAppData\fP (e.g. \fBC:\\Users\\Bactrian\\AppData\\Local\fP),
\fBRoamingAppData\fP (e.g. \fBC:\\Users\\Bactrian\\AppData\\Roaming\fP), or
\fBProgramData\fP (e.g. \fBC:\\ProgramData\fP) on Windows;
.IP 6.
\fBHOME/.ocamlinit\fP, if \fBHOME\fP is non-empty;

You can specify a different initialization file
by using the
.BI \-init " file"
option, and disable initialization files by using the
.B \-noinit
option.

Note that you can also use the
.B #use
directive to read phrases from a file.

.SH ENVIRONMENT VARIABLES
.TP
.B OCAMLTOP_UTF_8
When printing string values, non-ascii bytes (>0x7E) are printed as
decimal escape sequence if
.B OCAMLTOP_UTF_8
is set to false. Otherwise they are printed unescaped.
.TP
.B TERM
When printing error messages, the toplevel system
attempts to underline visually the location of the error. It
consults the TERM variable to determines the type of output terminal
and look up its capabilities in the terminal database.
.TP
.B XDG_CONFIG_HOME HOME XDG_CONFIG_DIRS
See
.B INITIALIZATION FILE
above.
.SH SEE ALSO
.BR ocamlc "(1), " ocamlopt "(1), " ocamlrun (1).
.br
.IR The\ OCaml\ user's\ manual ,
chapter "The toplevel system".


News
OCaml 4.10.0 (21 February 2020)
-------------------------------

- New best-fit allocator for the major heap
- Preliminary runtime work for OCaml multicore
- Immutable strings are now enforced at configuration time
- User-defined indexing operators for multidimensional arrays
- Coming soon: statmemprof, a new statistical memory profiler.
- The external API will be released next version.
- Various improvements to the manual
- More precise exhaustiveness check for GADTs
- Many bug fixes


OCaml 4.09.1 (18 March 2020)
----------------------------

Bug fixes.

OCaml 4.09.0 (18 September 2019)
--------------------------------

- New optimisations, in particular for affine functions in matches, for instance:

     type t = A | B | C
     let affine = function
       | A -> 4
       | B -> 3
       | C -> 2

- The `graphics` library was moved out of the compiler distribution.
- The `vmthread` library was removed.
- Support for compiler plugins was removed.
- Many bug fixes.

OCaml 4.08.1 (5 August 2019)
----------------------------

Bug fixes.

OCaml 4.08.0 (14 June 2019)
---------------------------

- Binding operators (let*, let+, and*, etc). They can be used to
  streamline monadic code.

- `open` now applies to arbitrary module expression in structures and
  to applicative paths in signatures.

- A new notion of (user-defined) "alerts" generalizes the deprecated
  warning.

- New modules in the standard library: Fun, Bool, Int, Option, Result.

- A significant number of new functions in Float, including FMA
  support, and a new Float.Array submodule.

- Source highlighting for errors and warnings in batch mode.

- Many error messages were improved.

- Improved AFL instrumentation for objects and lazy values.


OCaml 4.07.1 (4 October 2018)
-----------------------------

This release consists mostly of bug fixes. The most salient bugs were

- MPR#7820, GPR#1897: a bug in Array.of_seq (new in 4.07)
  (Thierry Martinez, review by Nicolás Ojeda Bär)

- MPR#7815, GPR#1896: crash in the major GC with the first-fit policy
  (Stephen Dolan and Damien Doligez, report by Joris Giovannangeli)

- MPR#7821, GPR#1908: the compiler loads more cmi, which breaks some builds
  (Jérémie Dimino, review by Gabriel Scherer)

- MPR#7833, GPR#1946: typechecking failure (regression) on large GADT matchings
  (Thomas Refis, report by Jerome Simeon, review by Jacques Garrigue)

See the detailed list of fixes at (Changes#4.07.1).


OCaml 4.07.0 (10 July 2018):
----------------------------

Some highlights of this release are:

-   The way the standard library modules are organized internally has
    changed (GPR#1010, by Jérémie Dimino):

    1. the `List` module (for example) is now named `Stdlib__list`
    2. a new Stdlib module contains a series of aliases
    such as `module List = Stdlib__list`
    3. the `Stdlib` module is implicitly opened when type-checking OCaml
    programs (as `Pervasives` previously was), so that `Stdlib.List` can be
    accessed as just `List`, as before.

    This should be invisible to most users, although it is possible that
    some tools show the `Stdlib.` or `Stdlib__` prefixes in
    messages. (You might want to report these situations as usability
    bugs.) The change prevents standard library modules from conflicting
    with end-user filenames (please avoid `stdlib.ml` and the
    `Stdlib__` prefix); we may introduce new standard library modules in
    the future with less fear of breaking user code. In particular,
    `Float` (GPR#1638, by Nicolás Ojeda Bär) and `Seq` (GPR#1002, by
    Simon Cruanes) modules have now been added to the standard library.

-   The error messages caused by various typing errors have been improved
    to be easier to understand, in particular for beginners.
    (GPR#1505, GPR#1510, by Arthur Charguéreau and Armaël Guéneau)

    For example,

        # while 1 do () done;;
                ^
        Error: This expression has type int but
               an expression was expected of type bool

    now adds the extra explanation

        because it is in the condition of a while-loop

-   Effort has been made to reduce the compilation time of flambda
    programs, and the size of the produced `.cmx` files when using
    the -Oclassic optimisation level.
    (GPR#1401, GPR#1455, GPR#1627, GPR#1665, by Pierre Chambart, Xavier
    Clerc, Fuyong Quah, and Leo White)

-   The HTML manual has benefited from various style improvements
    and should look visually nicer than previous editions.
    (GPR#1741, GPR#1757, GPR#1767 by Charles Chamberlain and steinuil)

    The new version of the manual can be consulted at
    <http://caml.inria.fr/pub/docs/manual-ocaml-4.07/>; see the
    previous version for comparison at
    <http://caml.inria.fr/pub/docs/manual-ocaml-4.06/>.

-   Since 4.01, it is possible to select a variant constructor or
    record field from a sub-module that is not opened in the current
    scope, if type information is available at the point of use. This
    now also works for GADT constructors.
    (GPR#1648, by Thomas Refis and Leo White)

-   The GC should handle the accumulation of custom blocks in the minor
    heap better; this solves some memory-usage issues observed by code
    which allocates a lot of small custom blocks, typically small bigarrays
    (GPR#1476, by Alain Frsich)

See also the detailed list of changes: (Changes#4.07.0).


OCaml 4.06.1 (16 Feb 2018):
---------------------------

This release consists mostly of bug fixes. The most salient bugs were

-   An incorrect compilation of pattern-matching in presence of
    extensible variant constructors (such as exceptions), that had been
    present for a long time.
    (GPR#1459, GPR#1538, by Luc Maranget, Thomas Refis and Gabriel Scherer)

-   An optimization of `not (x = y)` into `x <> y`, introduced in
    4.06.0, is incorrect on floating-point numbers in the `nan`
    case. (GPR#1470, by Leo White)

See the detailed list of fixes at (Changes#4.06.1).



man/ocaml.m
.\"**************************************************************************
.\"*                                                                        *
.\"*                                 OCaml                                  *
.\"*                                                                        *
.\"*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           *
.\"*                                                                        *
.\"*   Copyright 1996 Institut National de Recherche en Informatique et     *
.\"*     en Automatique.                                                    *
.\"*                                                                        *
.\"*   All rights reserved.  This file is distributed under the terms of    *
.\"*   the GNU Lesser General Public License version 2.1, with the          *
.\"*   special exception on linking described in the file LICENSE.          *
.\"*                                                                        *
.\"**************************************************************************
.\"
.TH OCAML 1

.SH NAME
ocaml \- The OCaml interactive toplevel

.SH SYNOPSIS
.B ocaml
[
.I options
]
[
.I object-files
]
[
.I script-file
]
.SH DESCRIPTION

The
.BR ocaml (1)
command is the toplevel system for OCaml,
that permits interactive use of the OCaml system through a
read-eval-print loop. In this mode, the system repeatedly reads OCaml
phrases from the input, then typechecks, compiles and evaluates
them, then prints the inferred type and result value, if any. The
system prints a # (hash) prompt before reading each phrase.

A toplevel phrase can span several lines. It is terminated by ;; (a
double-semicolon). The syntax of toplevel phrases is as follows.

The toplevel system is started by the command
.BR ocaml (1).
Phrases are read on standard input, results are printed on standard
output, errors on standard error. End-of-file on standard input
terminates
.BR ocaml (1).

If one or more
.I object-files
(ending in .cmo or .cma) are given, they are loaded silently before
starting the toplevel.

If a
.I script-file
is given, phrases are read silently from the file, errors printed on
standard error.
.BR ocaml (1)
exits after the execution of the last phrase.

.SH OPTIONS

The following command-line options are recognized by
.BR ocaml (1).
.TP
.B \-absname
Show absolute filenames in error messages.
.TP
.BI \-I \ directory
Add the given directory to the list of directories searched for
source and compiled files. By default, the current directory is
searched first, then the standard library directory. Directories added
with
.B \-I
are searched after the current directory, in the order in which they
were given on the command line, but before the standard library
directory.
.IP
If the given directory starts with
.BR + ,
it is taken relative to the
standard library directory. For instance,
.B \-I\ +compiler-libs
adds the subdirectory
.B compiler-libs
of the standard library to the search path.
.IP
Directories can also be added to the search path once the toplevel
is running with the
.B #directory
directive.
.TP
.BI \-init \ file
Load the given file instead of the default initialization file.
See the "Initialization file" section below.
.TP
.B \-labels
Labels are not ignored in types, labels may be used in applications,
and labelled parameters can be given in any order.  This is the default.
.TP
.B \-no\-app\-funct
Deactivates the applicative behaviour of functors. With this option,
each functor application generates new types in its result and
applying the same functor twice to the same argument yields two
incompatible structures.
.TP
.B \-noassert
Do not compile assertion checks.  Note that the special form
.B assert\ false
is always compiled because it is typed specially.
.TP
.B \-noinit
Do not load any initialization file.
See the "Initialization file" section below.
.TP
.B \-nolabels
Ignore non-optional labels in types. Labels cannot be used in
applications, and parameter order becomes strict.
.TP
.B \-noprompt
Do not display any prompt when waiting for input.
.TP
.B \-nopromptcont
Do not display the secondary prompt when waiting for continuation lines in
multi-line inputs.  This should be used e.g. when running
.BR ocaml (1)
in an
.BR emacs (1)
window.
.TP
.B \-nostdlib
Do not include the standard library directory in the list of
directories searched for source and compiled files.
.TP
.BI \-open \ module
Opens the given module before starting the toplevel. If several
.B \-open
options are given, they are processed in order, just as if
the statements open! module1;; ... open! moduleN;; were input.
.TP
.BI \-ppx \ command
After parsing, pipe the abstract syntax tree through the preprocessor
.IR command .
The module
.BR Ast_mapper (3)
implements the external interface of a preprocessor.
.TP
.B \-principal
Check information path during type-checking, to make sure that all
types are derived in a principal way.  When using labelled arguments
and/or polymorphic methods, this flag is required to ensure future
versions of the compiler will be able to infer types correctly, even
if internal algorithms change.
All programs accepted in
.B \-principal
mode are also accepted in the
default mode with equivalent types, but different binary signatures,
and this may slow down type checking; yet it is a good idea to
use it once before publishing source code.
.TP
.B \-rectypes
Allow arbitrary recursive types during type-checking.  By default,
only recursive types where the recursion goes through an object type
are supported.
.TP
.B \-safe\-string
Enforce the separation between types
.BR string \ and\  bytes ,
thereby making strings read-only. This is the default.
.TP
.B \-short\-paths
When a type is visible under several module-paths, use the shortest
one when printing the type's name in inferred interfaces and error and
warning messages.
.TP
.B \-stdin
Read the standard input as a script file rather than starting an
interactive session.
.TP
.B \-strict\-sequence
Force the left-hand part of each sequence to have type unit.
.TP
.B \-unboxed\-types
When a type is unboxable (i.e. a record with a single argument or a
concrete datatype with a single constructor of one argument) it will
be unboxed unless annotated with
.BR [@@ocaml.boxed] .
.TP
.B \-no-unboxed\-types
When a type is unboxable  it will be boxed unless annotated with
.BR [@@ocaml.unboxed] .
This is the default.
.TP
.B \-unsafe
Turn bound checking off on array and string accesses (the
.BR v.(i) and s.[i]
constructs). Programs compiled with
.B \-unsafe
are therefore slightly faster, but unsafe: anything can happen if the program
accesses an array or string outside of its bounds.
.TP
.B \-unsafe\-string
Identify the types
.BR string \ and\  bytes ,
thereby making strings writable.
This is intended for compatibility with old source code and should not
be used with new software.
.TP
.B \-version
Print version string and exit.
.TP
.B \-vnum
Print short version number and exit.
.TP
.B \-no\-version
Do not print the version banner at startup.
.TP
.BI \-w \ warning\-list
Enable or disable warnings according to the argument
.IR warning-list .
See
.BR ocamlc (1)
for the syntax of the
.I warning\-list
argument.
.TP
.BI \-warn\-error \ warning\-list
Mark as fatal the warnings described by the argument
.IR warning\-list .
Note that a warning is not triggered (and does not trigger an error) if
it is disabled by the
.B \-w
option.  See
.BR ocamlc (1)
for the syntax of the
.I warning\-list
argument.
.TP
.BI \-color \ mode
Enable or disable colors in compiler messages (especially warnings and errors).
The following modes are supported:

.B auto
use heuristics to enable colors only if the output supports them (an
ANSI-compatible tty terminal);

.B always
enable colors unconditionally;

.B never
disable color output.

The environment variable "OCAML_COLOR" is considered if \-color is not
provided. Its values are auto/always/never as above.

If \-color is not provided, "OCAML_COLOR" is not set and the environment
variable "NO_COLOR" is set, then color output is disabled. Otherwise,
the default setting is
.B auto,
and the current heuristic
checks that the "TERM" environment variable exists and is
not empty or "dumb", and that isatty(stderr) holds.

.TP
.BI \-error\-style \ mode
Control the way error messages and warnings are printed.
The following modes are supported:

.B short
only print the error and its location;

.B contextual
like "short", but also display the source code snippet corresponding
to the location of the error.

The default setting is
.B contextual.

The environment variable "OCAML_ERROR_STYLE" is considered if
\-error\-style is not provided. Its values are short/contextual as
above.

.TP
.B \-warn\-help
Show the description of all available warning numbers.
.TP
.BI \- \ file
Use
.I file
as a script file name, even when it starts with a hyphen (-).
.TP
.BR \-help \ or \ \-\-help
Display a short usage summary and exit.

.SH INITIALIZATION FILE

When
.BR ocaml (1)
is invoked, it will read phrases from an initialization file before
giving control to the user. The default file is
.B .ocamlinit
in the current directory if it exists, otherwise
.B XDG_CONFIG_HOME/ocaml/init.ml
according to the XDG base directory specification lookup if it exists (on
Windows this is skipped), otherwise
.B .ocamlinit
in the user's home directory (
.B HOME
variable).
You can specify a different initialization file
by using the
.BI \-init \ file
option, and disable initialization files by using the
.B \-noinit
option.

Note that you can also use the
.B #use
directive to read phrases from a file.

.SH ENVIRONMENT VARIABLES
.TP
.B OCAMLTOP_UTF_8
When printing string values, non-ascii bytes (>0x7E) are printed as
decimal escape sequence if
.B OCAMLTOP_UTF_8
is set to false. Otherwise they are printed unescaped.
.TP
.B TERM
When printing error messages, the toplevel system
attempts to underline visually the location of the error. It
consults the TERM variable to determines the type of output terminal
and look up its capabilities in the terminal database.
.TP
.B XDG_CONFIG_HOME HOME
.B .ocamlinit
lookup procedure (see above).
.SH SEE ALSO
.BR ocamlc (1), \ ocamlopt (1), \ ocamlrun (1).
.br
.IR The\ OCaml\ user's\ manual ,
chapter "The toplevel system".
